### PR TITLE
customize: Add documentation for configuration variables

### PR DESCRIPTION
**Pull request by sakshamarora1**

### CODE REVIEW

File: docs/customize/configure.md [hunk 1]
```diff
@@ -0,0 +1,621 @@
+# Configuration Variables
+
+This document provides a comprehensive list and explanation of the configuration variables available in the `invenio-app-ils` backend. These variables can be set in the configuration file(s) to customize the behavior of your Invenio ILS application.
+
+## REST Configurations
+
+### `ILS_RECORDS_REST_ENDPOINTS`
```
Comment (kpsherva) [inline docs/customize/configure.md:7]: I wouldn't add this one, I think it is too advanced, maybe we need some "advanced"  config section or remove completely

File: docs/customize/configure.md [hunk 2]
```diff
@@ -0,0 +1,621 @@
...
+```
+
+## Circulation Configuration
+
+### `CIRCULATION_LOAN_TRANSITIONS`
+
+A dictionary defining the possible transitions for loans.
+
+```python
+CIRCULATION_LOAN_TRANSITIONS = {
```
Comment (kpsherva) [inline docs/customize/configure.md:44]: this I would describe a bit more as a use case: for example if you need to add a loan in transit state or skip a step in the loan circulation, then you need to modify this variable. Also a bit advanced config.

File: docs/customize/configure.md [hunk 3]
```diff
@@ -0,0 +1,621 @@
...
+
+For detailed documentation for circulation sub-module, refer to [invenio-circulation](https://invenio-circulation.readthedocs.io/en/latest/).
+
+## Debug Configuration
+
+### DEBUG (True/False)
+
+Enable or disable debugging mode.
+
+### DEBUG_TB_ENABLED (True/False)
```
Comment (kpsherva) [inline docs/customize/configure.md:61]: this one does not work, to remove

File: docs/customize/configure.md [hunk 4]
```diff
@@ -0,0 +1,621 @@
...
+
+Enable or disable the debug toolbar.
+
+### DEBUG_TB_INTERCEPT_REDIRECTS (True/False)
```
Comment (kpsherva) [inline docs/customize/configure.md:65]: same as above

File: docs/customize/configure.md [hunk 5]
```diff
@@ -0,0 +1,621 @@
...
+Enable or disable intercepting redirects in the debug toolbar.
+
+## Rate Limiting Configuration
+
+### RATELIMIT_STORAGE_URL
+
+Storage URL for the rate limiter.
+
+```python
+RATELIMIT_STORAGE_URL = "redis://localhost:6379/3"
```
Comment (kpsherva) [inline docs/customize/configure.md:76]: this is more "for website administrators" section of configuration, not features related one

File: docs/customize/configure.md [hunk 6]
```diff
@@ -0,0 +1,621 @@
...
+```
+
+## Internationalization (I18N) Configuration
+
+### BABEL_DEFAULT_LANGUAGE
+
+Default language setting.
+
+```python
+BABEL_DEFAULT_LANGUAGE = "en"
```
Comment (kpsherva) [inline docs/customize/configure.md:102]: I would add a disclaimer that for now the software is available only in english

File: docs/customize/configure.md [hunk 7]
```diff
@@ -0,0 +1,621 @@
...
+
+Default timezone setting.
+
+```python
+BABEL_DEFAULT_TIMEZONE = "Europe/Zurich"
+```
+
+## Notifications Configuration
+
+### ILS_NOTIFICATIONS_BACKENDS_BUILDER
```
Comment (kpsherva) [inline docs/customize/configure.md:115]: this variable and the next are advanced config also

File: docs/customize/configure.md [hunk 8]
```diff
@@ -0,0 +1,621 @@
...
+    "invenio_app_ils.notifications.messages:notification_msg_builder"
+)
+```
+
+### ILS_NOTIFICATIONS_TEMPLATES
+
+Override default global common templates.
+
+```python
+ILS_NOTIFICATIONS_TEMPLATES = {"footer": "footer.html"}
```
Comment (kpsherva) [inline docs/customize/configure.md:140]: I would give more possible key value pairs here to show what is possible

File: docs/customize/configure.md [hunk 9]
```diff
@@ -0,0 +1,621 @@
...
+```
+
+### ILS_NOTIFICATIONS_MSG_BUILDER_DOCUMENT_REQUEST
+
+Notification message creator for document requests notifications.
+
+```python
+ILS_NOTIFICATIONS_MSG_BUILDER_DOCUMENT_REQUEST = "invenio_app_ils.document_requests.notifications.messages:notification_document_request_msg_builder"
```
Comment (kpsherva) [inline docs/customize/configure.md:148]: this one is also advanced config

File: docs/customize/configure.md [hunk 10]
```diff
@@ -0,0 +1,621 @@
...
+
+Override default document requests templates.
+
+```python
+ILS_NOTIFICATIONS_TEMPLATES_DOCUMENT_REQUEST = {}
+```
+
+### ILS_NOTIFICATIONS_FILTER_DOCUMENT_REQUEST
+
+Function to select and filter which notifications should be sent.
```
Comment (kpsherva) [inline docs/customize/configure.md:161]:
```suggestion
Function to select and filter which notifications should be sent. Possibility to filter out some emails, based on your needs.
```

File: docs/customize/configure.md [hunk 11]
```diff
@@ -0,0 +1,621 @@
...
+
+When `ILS_MAIL_ENABLE_TEST_RECIPIENTS` is True, all emails are sent here.
+
+```python
+ILS_MAIL_NOTIFY_TEST_RECIPIENTS = ["john.doe@example.com"]
+```
+
+## Assets Configuration
+
+### COLLECT_STORAGE
```
Comment (kpsherva) [inline docs/customize/configure.md:211]: advanced configuration

File: docs/customize/configure.md [hunk 12]
```diff
@@ -0,0 +1,621 @@
...
+        "args": [("record-view-agg", "file-download-agg")],
+    },
+}
+```
+
+## Feature Configurations
+
+### EXTEND_LOANS_LOCATION_UPDATED (True/False)
+
+Flag to indicate whether to start a task to update active loans when closures of a location are updated.
```
Comment (kpsherva) [inline docs/customize/configure.md:263]: I think this needs a bit more context to explain what is happening

File: docs/customize/configure.md [hunk 13]
```diff
@@ -0,0 +1,621 @@
...
+
+Time at which the task to extend loans should be run.
+
+```python
+EXTEND_LOANS_SCHEDULE_TIME = datetime.time(2)
+```
+
+### ILS_SELF_CHECKOUT_ENABLED (True/False)
+
+Enable or disable Self-Checkout feature endpoint.
```
Comment (kpsherva) [inline docs/customize/configure.md:275]:
```suggestion
Enable or disable Self-Checkout feature.
```

File: docs/customize/configure.md [hunk 14]
```diff
@@ -0,0 +1,621 @@
...
+
+## Database Configuration
+
+### SQLALCHEMY_DATABASE_URI
+
+Database URI including user and password.
+
+```python
```
Comment (kpsherva) [inline docs/customize/configure.md:283]: I think it should be clustered with other website admin/developer variables

File: docs/customize/configure.md [hunk 15]
```diff
@@ -0,0 +1,621 @@
...
+
+Enable or disable CORS support for REST APIs.
+
+### CORS_SEND_WILDCARD (True/False)
+
+Allow or disallow credentials to be sent with CORS requests.
+
+### CORS_SUPPORTS_CREDENTIALS (True/False)
+
+Exable or disable supporting credentials for CORS requests.
```
Comment (kpsherva) [inline docs/customize/configure.md:315]: these are "security variables", we should add a warning that it is not advised to modify them

File: docs/customize/configure.md [hunk 16]
```diff
@@ -0,0 +1,621 @@
...
+
+## Flask Configuration
+
+### SECRET_KEY
```
Comment (kpsherva) [inline docs/customize/configure.md:319]: this is also security variable, we need a note indicating that this needs to be changed to a secure key when website goes to production

File: docs/customize/configure.md [hunk 17]
```diff
@@ -0,0 +1,621 @@
...
+
+Set cookie with the secure flag by default.
+
+### SESSION_COOKIE_SAMESITE
+
+Set SameSite attribute for session cookies.
+
+```python
+SESSION_COOKIE_SAMESITE = "Lax"
+```
```
Comment (kpsherva) [inline docs/customize/configure.md:341]: also "security related" variables

File: docs/customize/configure.md [hunk 18]
```diff
@@ -0,0 +1,621 @@
...
+
+### SPA_HOST
+
+Host URL for Single Page Application.
+
+```python
+SPA_HOST = "https://127.0.0.1:3000"
+```
+
+### SPA_PATHS
```
Comment (kpsherva) [inline docs/customize/configure.md:359]: candidates for site admin/devops variables section

File: docs/customize/configure.md [hunk 19]
```diff
@@ -0,0 +1,621 @@
...
+
+Paths and placeholders for Single Page Application routes.
+
+```python
+SPA_PATHS = dict(
```
Comment (kpsherva) [inline docs/customize/configure.md:364]: I would drop this one from docs, I am not sure what will happen if it is modified

File: docs/customize/configure.md [hunk 20]
```diff
@@ -0,0 +1,621 @@
...
+```python
+RECORDS_REST_VOCAB_MAX_RESULT_WINDOW = 500
+```
+
+### PIDSTORE_RECID_FIELD
+
+Field name in the PID store for the record identifier.
+
+```python
+PIDSTORE_RECID_FIELD = "pid"
```
Comment (kpsherva) [inline docs/customize/configure.md:398]: I would drop this one

File: docs/customize/configure.md [hunk 21]
```diff
@@ -0,0 +1,621 @@
...
+```
+
+### REST_MIMETYPE_QUERY_ARG_NAME
+
+Name of the URL argument to choose response serializer.
+
+```python
+REST_MIMETYPE_QUERY_ARG_NAME = "format"
```
Comment (kpsherva) [inline docs/customize/configure.md:406]: I would drop this one as well

File: docs/customize/configure.md [hunk 22]
```diff
@@ -0,0 +1,621 @@
...
+### RECORDS_REST_ENDPOINTS
+
+Dictionary defining the endpoints for Records REST API.
+
+```python
+RECORDS_REST_ENDPOINTS = {
+    "docid": {
+        "pid_type": DOCUMENT_PID_TYPE,
+        "pid_minter": DOCUMENT_PID_MINTER,
+        "pid_fetcher": DOCUMENT_PID_FETCHER,
```
Comment (kpsherva) [inline docs/customize/configure.md:418]: to drop these

File: docs/customize/configure.md [hunk 23]
```diff
@@ -0,0 +1,621 @@
...
+        },
+        "search_serializers": {
+            "application/json": "invenio_app_ils.literature.serializers:json_v1_search",
+            "text/csv": "invenio_app_ils.literature.serializers:csv_v1_search",
+        },
+        "search_serializers_aliases": {
+            "csv": "text/csv",
+            "json": "application/json",
+        },
+        "list_route": "/documents/",
```
Comment (kpsherva) [inline docs/customize/configure.md:437]: to drop

File: docs/customize/configure.md [hunk 24]
```diff
@@ -0,0 +1,621 @@
...
+        "item_route": "/documents/<{0}:pid_value>".format(doc_pid_converter()),
```
Comment (kpsherva) [inline docs/customize/configure.md:438]: to drop

File: docs/customize/configure.md [hunk 25]
```diff
@@ -0,0 +1,621 @@
...
+        "default_media_type": "application/json",
+        "max_result_window": RECORDS_REST_MAX_RESULT_WINDOW,
+        "error_handlers": {},
+        "read_permission_factory_imp": record_read_permission_factory,
+        "list_permission_factory_imp": allow_all,
+        "create_permission_factory_imp": backoffice_permission,
+        "update_permission_factory_imp": backoffice_permission,
+        "delete_permission_factory_imp": backoffice_permission,
```
Comment (kpsherva) [inline docs/customize/configure.md:446]: I would leave this and explain that they are used to implement permission policies for specific endpoints

File: docs/customize/configure.md [hunk 26]
```diff
@@ -0,0 +1,621 @@
...
+            "application/json": "invenio_app_ils.literature.serializers:json_v1_search",
+            "text/csv": "invenio_app_ils.literature.serializers:csv_v1_search",
+        },
+        "search_serializers_aliases": {
+            "csv": "text/csv",
+            "json": "application/json",
+        },
+        "list_route": "/documents/",
+        "item_route": "/documents/<{0}:pid_value>".format(doc_pid_converter()),
+        "default_media_type": "application/json",
```
Comment (kpsherva) [inline docs/customize/configure.md:439]: to drop

File: docs/customize/configure.md [hunk 27]
```diff
@@ -0,0 +1,621 @@
...
+        "max_result_window": RECORDS_REST_MAX_RESULT_WINDOW,
+        "error_handlers": {},
```
Comment (kpsherva) [inline docs/customize/configure.md:441]: to drop

File: docs/customize/configure.md [hunk 28]
```diff
@@ -0,0 +1,621 @@
...
+            "application/json": "invenio_app_ils.literature.serializers:json_v1_response"
+        },
+        "search_serializers": {
+            "application/json": "invenio_app_ils.literature.serializers:json_v1_search",
+            "text/csv": "invenio_app_ils.literature.serializers:csv_v1_search",
+        },
+        "search_serializers_aliases": {
+            "csv": "text/csv",
+            "json": "application/json",
+        },
```
Comment (kpsherva) [inline docs/customize/configure.md:436]: to drop

File: docs/customize/configure.md [hunk 29]
```diff
@@ -0,0 +1,621 @@
...
+
+```python
+RECORDS_REST_ENDPOINTS = {
+    "docid": {
+        "pid_type": DOCUMENT_PID_TYPE,
+        "pid_minter": DOCUMENT_PID_MINTER,
+        "pid_fetcher": DOCUMENT_PID_FETCHER,
+        "search_class": DocumentSearch,
+        "record_class": Document,
+        "indexer_class": DocumentIndexer,
```
Comment (kpsherva) [inline docs/customize/configure.md:421]: I would leave these and explain that they can be modified if customisation is needed on the Record class behaviour or search class or indexer, but these require some advanced developer experience

File: docs/customize/configure.md [hunk 30]
```diff
@@ -0,0 +1,621 @@
...
+    "vocid": {...},
+    "litid": {...},
+}
+```
+
+For a detailed configuration reference, refer to [invenio-records-rest](https://invenio-records-rest.readthedocs.io/en/latest/).
+
+### RECORDS_REST_SORT_OPTIONS
+
+Dictionary defining sort options for Records REST endpoints.
```
Comment (kpsherva) [inline docs/customize/configure.md:463]:
```suggestion
Dictionary defining sort options for Records REST endpoints. Sort options ser here will be displayed in the search pages.
```

File: docs/customize/configure.md [hunk 31]
```diff
@@ -0,0 +1,621 @@
...
+
+### OPENDEFINITION_JSONRESOLVER_HOST
+
+Host URL for opendefinition license for jsonresolvers.
+
+### FILES_REST_PERMISSION_FACTORY
+
+Factory function for defining permissions for file upload.
+
+### ILS_RECORDS_EXPLICIT_PERMISSIONS_ENABLED (True/False)
```
Comment (kpsherva) [inline docs/customize/configure.md:587]: To drop

File: docs/customize/configure.md [hunk 32]
```diff
@@ -0,0 +1,621 @@
...
+When enabled, it allows to define explicit permissions for each record to
+provide read access to specific users or roles.
+When disabled, it will avoid checking for user ids and roles on each search
+query and record fetch.
+
+### ILS_LITERATURE_COVER_URLS_BUILDER
+
+Default implementation for building cover urls in document serializer.
+
+### ILS_RECORDS_METADATA_NAMESPACES
```
Comment (kpsherva) [inline docs/customize/configure.md:600]: to drop

File: docs/customize/configure.md [hunk 33]
```diff
@@ -0,0 +1,621 @@
...
+
+Namespaces for fields added to the metadata schema.
+
+### ILS_RECORDS_METADATA_EXTENSIONS
+
+Fields added to the metadata schema.
```
Comment (kpsherva) [inline docs/customize/configure.md:606]: we need an example on how to configure

File: docs/customize/configure.md [hunk 34]
```diff
@@ -0,0 +1,621 @@
...
+
+### ILS_PATRON_ANONYMOUS_CLASS and ILS_PATRON_SYSTEM_AGENT_CLASS
```
Comment (kpsherva) [inline docs/customize/configure.md:608]: to drop, I think it is quite unlikely that someone will change it

File: docs/customize/configure.md [hunk 35]
```diff
@@ -0,0 +1,621 @@
+# Configuration Variables
+
+This document provides a comprehensive list and explanation of the configuration variables available in the `invenio-app-ils` backend. These variables can be set in the configuration file(s) to customize the behavior of your Invenio ILS application.
```
Comment (ntarocco) [inline docs/customize/configure.md:3]: Same remark/comment as in the other PR. I would stick to `InvenioILS` (one word) and avoid modules name.
Or, if you want to explicit mention `invenio-app-ils`, then I would explain that is the module (and not really the backend in the true sense of the term) with a link:
```suggestion
This document provides a comprehensive list and explanation of the configuration variables available in the [invenio-app-ils](https://github.com/inveniosoftware/invenio-app-ils) module. These variables can be set in the configuration file(s) to customize the behavior of your InvenioILS application.
```

**Review by ntarocco — COMMENTED** [review]
Agree with Karolina's comment, I am wondering if listing everything clarifies or confuses the reader. I would focus on the variables that you would really touch/change in a real instance, for example the e-mail templates, or the DB connection.
Not for now: for the rest, I think they should be explained in the context of a real use case or example.

**Pull request merged**